### PR TITLE
Changed workflow to include matrix for multiple flavors of linux 

### PR DESCRIPTION
**Pull request by MigeljanImeri**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,46 +11,66 @@ jobs:
...
     - name: make
       run: make -C ${{github.workspace}}/build
     - name: build deb
       run: make -C ${{github.workspace}}/build package
     - name: install deb
-      run: sudo apt install -y ${{github.workspace}}/build/xdd-*.deb
+      run: apt install -y ${{github.workspace}}/build/xdd-*.deb
     - name: run tests
       run: ctest --output-on-failure
       working-directory: ${{github.workspace}}/build
```
Comment (calccrypto) [inline .github/workflows/ci.yml]: Can probably merge this into `test` with 

```
matrix:
    pkg: [ "On", "Off" ]
```

and reduce duplicate steps

File: CMakeLists.txt [hunk 2]
```diff
@@ -19,6 +19,9 @@ find_program(PYTHON REQUIRED NAMES python3 python)
 
 # Threads are required
 find_package(Threads REQUIRED)
+if (Threads_FOUND)
```
Comment (calccrypto) [inline CMakeLists.txt]: This is not necessary. `find_package(THREADS)` already has `REQUIRED`.

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +19,9 @@ find_program(PYTHON REQUIRED NAMES python3 python)
...
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
```
Comment (calccrypto) [inline CMakeLists.txt]: This is not necessary. `find_package(THREADS)` sets compiler and linker flags with `target_link_libraries`. 
https://stackoverflow.com/a/49769972

File: CMakeLists.txt [hunk 4]
```diff
@@ -19,6 +19,9 @@ find_program(PYTHON REQUIRED NAMES python3 python)
 
 # Threads are required
 find_package(Threads REQUIRED)
```
Comment (calccrypto) [inline CMakeLists.txt:19]: Might want to have `set(THREADS_PREFER_PTHREAD_FLAG ON)`.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -11,46 +11,66 @@ jobs:
     name: build
 
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        #img: [ "centos:7", "centos:8", "rockylinux:8", "ubuntu:20.04", "ubuntu:22.04" ]
+        img: ["ubuntu:20.04", "ubuntu:22.04", "centos:8" ]
```
Comment (calccrypto) [inline .github/workflows/ci.yml:16]: Recommend adding to matrix `infiniband: ["On", "Off"]` and `numactl: ["On", "Off"]` to test different combinations of libraries.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -11,44 +11,73 @@ jobs:
...
       run: make -C ${{github.workspace}}/build
     - name: make install
-      run: sudo make -C ${{github.workspace}}/build install
+      run: make -C ${{github.workspace}}/build install
     - name: run tests
-      run: ctest --output-on-failure
-      working-directory: ${{github.workspace}}/build
+      run:  | 
+            cd ${{github.workspace}}/build 
+            ctest --output-on-failure
```
Comment (calccrypto) [inline .github/workflows/ci.yml]: Why change this?

Comment (MigeljanImeri) [reply .github/workflows/ci.yml]: I ran into errors using the working-directory syntax in the workflow. I am not too sure why it was causing issues, but swapping to this fixed it. 
image

Comment (calccrypto) [reply .github/workflows/ci.yml]: Ok. Just double checking that that was the issue. The github actions variables are weird.

File: CMakeLists.txt [hunk 7]
```diff
@@ -57,7 +58,7 @@ if (WITH_NUMA)
 
     set(CPACK_RPM_PACKAGE_REQUIRES "${CPACK_RPM_PACKAGE_REQUIRES}, numactl-libs")
     list(APPEND CPACK_DEBIAN_PACKAGE_DEPENDS "libnuma1")
-
+
```
Comment (calccrypto) [inline CMakeLists.txt]: Extra spaces

**Review by calccrypto — APPROVED** [review]

**Review by bwatkinson — CHANGES_REQUESTED** [review]
So overall this looks good to me. However, I would suggest we don't do the toggle on/off for building the deb/rpms. Let's just build those always. The `ctest` that is being run is doing so from the `bin` executables in the build directory anyways. It is good enough to just make sure the packages build before running the tests.

Also, can we maybe squash all these commits down into a single commit?

File: CMakeLists.txt (hunk 7 above)
Comment (MigeljanImeri) [reply CMakeLists.txt]: Removed the extra space.

File: contrib/CI/ubuntu.sh [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash 
+
+set -e
+
+#https://grigorkh.medium.com/fix-tzdata-hangs-docker-image-build-cdb52cc3360d
+#Added because github runner would get stuck on interactive option when installing 
+#tzdata
+TZ="America/New_York"
```
Comment (calccrypto) [inline contrib/CI/ubuntu.sh]: Denver

Comment (bwatkinson) [reply contrib/CI/ubuntu.sh]: @MigeljanImeri after you adjust this, feel free to merge it.

**Review by bwatkinson — APPROVED** [review]

**Pull request merged**